test/DynamoCoreTests/Core/NotificationObjectTests.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using Dynamo.Core;
+using NUnit.Framework;
+
+namespace Dynamo.Tests.Core
+{
+    /// <summary>
+    /// Test class to test the NotificationObject class
+    /// </summary>
+    [TestFixture]
+    public class NotificationObjectTests : DynamoModelTestBase
+    {
+        private NotificationObjectTestingSubclass notificationObject;
+
+        /// <summary>
+        /// Tests initialization
+        /// </summary>
+        [SetUp]
+        public void Init()
+        {
+            notificationObject = new NotificationObjectTestingSubclass();
+            notificationObject.RaisePropertyChangedCalled = false;
+        }
+
+        /// <summary>
+        /// RaisePropertyChanged method testing
+        /// </summary>
+        [Test]
+        [Category("UnitTests")]
+        public void RaisePropertyChangedTest()
+        {
+            notificationObject.TestPropertySingle = !notificationObject.TestPropertySingle;
+            Assert.IsTrue(notificationObject.RaisePropertyChangedCalled);
+            
+            notificationObject.RaisePropertyChangedCalled = false; //Reset flag
+
+            notificationObject.TestPopertyMultiple = !notificationObject.TestPopertyMultiple;
+            Assert.IsTrue(notificationObject.RaisePropertyChangedCalled);
+            
+            notificationObject.RaisePropertyChangedCalled = false; //Reset flag
+
+            //The RaisePropertyChanged method that receives a string[] throws an ArgumentNullException if the array is null;
+            Assert.Throws<ArgumentNullException>(() => notificationObject.TestPropertyNull = !notificationObject.TestPropertyNull);
+            Assert.IsFalse(notificationObject.RaisePropertyChangedCalled);
+        }
+
+        /// <summary>
+        /// Derivate class from NotificationObject, designed to help in its testing
+        /// </summary>
+        private class NotificationObjectTestingSubclass : NotificationObject
+        {
+            private bool _testProperty;
+
+            /// <summary>
+            /// Used to test the RaisePopertyChanged method that receives a single property
+            /// </summary>
+            public bool TestPropertySingle
```

Comment:
Would you use `internal` for these properties?

Reply:
Done, I also made the constructor of the utility class and the flag internal.